leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/MessageStore.java

```diff
@@ -55,16 +57,16 @@
     boolean isEmpty(String endpoint);
 
     /**
-     * deletes all queued request for a given client endpoint.
-     *
-     * @param endpoint client's endpoint
-     */
-    void removeAll(String endpoint);
-
-    /**
      * Deletes the first request from message queue for the given client.
      *
      * @param endpoint client endpoint name
      */
     void deleteFirst(String endpoint);
+
+    /**
+     * 
+     * @param endpoint
+     * @return
+     */
+    List<QueuedRequest> removeAll(String endpoint);
```

Comment:
Actually javadoc is not complete here. I removed this method first and again added it with a different signature.